cerk_port_mqtt_mosquitto/src/lib.rs

```diff
@@ -0,0 +1,165 @@
+/*!
+
+This is a package for [CERK](https://github.com/ce-rust/cerk).
+CERK is an open source [CloudEvents](https://github.com/cloudevents/spec) Router written in Rust with a MicroKernel architecture.
+
+# Introduction
+
+CERK lets you route your [CloudEvents](https://github.com/cloudevents/spec) between different different ports.
+Ports are transport layer bindings over which CloudEvents can be exchanged.
+It is built with modularity and portability in mind.
+
+# Components
+
+CERK comes with a couple of prefabricated components, but implementing custom components is easy.
+
+A good overview is provided on [GitHub](https://github.com/ce-rust/cerk/).
+
+# This Component: MQTT Port using libmosquitto
+
+This port publishes and/or subscribe CloudEvents to/from an MQTT topic.
+
+The port is implemented with a fork of the [Mosquitto Client](https://docs.rs/mosquitto-client/0.1.5/mosquitto_client/)
+and sends and receives messages according to the CloudEvents specification (see [MQTT Protocol Binding for CloudEvents v1.0](https://github.com/cloudevents/spec/blob/v1.0/mqtt-protocol-binding.md)).
+
+The reason we are currently using a fork is that the version published on crates.io does currently not support the latest version of libmosquitto.
+Because we proposed a [change to libmostquitto](https://github.com/eclipse/mosquitto/pull/1932) to better support our usecase we need to use the headerfiles of latest version of libmosquitto.
+The goal is to used the published version at some point because otherwise this port can't be published on crates.io (see https://github.com/ce-rust/cerk/issues/88).
+
+
```

Comment:
please note that a specific library has to be added in order to use the port.
May refer to the test docker-compose.